pkg/generators/golang/builders_generator.go

```diff
@@ -247,19 +247,28 @@ func (g *BuildersGenerator) generateStructBuilderSource(typ *concepts.Type) {
 		{{ if .Type.IsClass }}
 			// Link sets the flag that indicates if this is a link.
 			func (b *{{ $builderName }}) Link(value bool) *{{ $builderName }} {
+				if len(b.fieldSet_) == 0 {
+					b.fieldSet_ = make([]bool, {{ fieldSetSize .Type }})
+				}
 				b.fieldSet_[0] = true
 				return b
 			}
 
 			// ID sets the identifier of the object.
 			func (b *{{ $builderName }}) ID(value string) *{{ $builderName }} {
+				if len(b.fieldSet_) == 0 {
+					b.fieldSet_ = make([]bool, {{ fieldSetSize .Type }})
+				}
 				b.id = value
 				b.fieldSet_[1] = true
 				return b
 			}
 
 			// HREF sets the link to the object.
 			func (b *{{ $builderName }}) HREF(value string) *{{ $builderName }} {
+				if len(b.fieldSet_) == 0 {
+					b.fieldSet_ = make([]bool, {{ fieldSetSize .Type }})
```

Comment:
What is the difference between `.Type` and `$.Type`?

Reply:
`.Type` refers to the current context in the template pipeline. When you're inside a `{{ range .Type.Attributes }}` loop, .Type refers to the current attribute's type.

`$.Type` refers to the root context (the dollar sign `$` represents the root context). This always refers to the original Type object that was passed to the template, regardless of where you are in nested loops or contexts.

Reply:
wait, I think you spotted a bug here...